Pipes - Angela - solar_system.rb

```diff
@@ -0,0 +1,377 @@
+########################################
+# Wave 3
+########################################
+
+# Goals
+  # 1. user can select a planet
+  # 2. user can view info about planet
+  # 3. user can add their own planet
+
+
+# Wave three is partially complete because it took me some time to understand the fundamentalls.
+# please provide some good feedback on my existing work.
+
+class Planet
+  # Line below is an attribute reader with all attributes in one method.
+  attr_reader :name, :volume, :solid, :diameter, :orbit_time
+
+
+  def initialize(name, volume, diameter, solid, orbit_time)
+    @name = name
+    @volume = volume
+    @solid = solid
+    @diameter = diameter
+    @orbit_time = orbit_time
+  end
+
+  def planet_output
+    planet_characteristics_ouput = "
+    Your planet's name = #{@name}.
+    Your planet's volume = #{@volume}.
+    Is it solid? #{@solid}.
+    Its diameter is #{@diameter}.
+    It takes #{@orbit_time} to orbit its sun."
+    return planet_characteristics_ouput
+  end
+
+  def planets_name
+    list_planet_names = "#{@name}\n"
+
+    return list_planet_names
+  end
+
+  def planet_detail
+    planet_details = "
+    #{@name}'s volume = #{@volume}.
+    Is it solid? #{@solid}.
+    #{@name} diameter is #{@diameter}.
+    It takes #{@orbit_time} to orbit its sun."
+    return planet_details
+  end
+
+  def name_of_planet
+    return @name
+  end
+end # end of planet class
+
+
+class SolarSystem
+  # attr_writer :name
+  def initialize(planet_object_collection)
+    # @name
+    @solar_system = planet_object_collection
+  end
+
+  # syntax used to override the existing characteristic
+  # def name=(name)
+  #   @name = name
+  # end
+
+  def add_planet_object(new_planet_object)
+    @solar_system.push(new_planet_object)
+  end
+
+  # Outputs all planet's details
+  def return_solar_system_info
+    solar_system_output = ""
+    @solar_system.each do |planet_object| # planet_object represents a new planet instance
+      #add to planet_output string
+      solar_system_output += planet_object.planet_output
+    end
+    return solar_system_output
+  end
+
+  # makes list of planet names for user
+  def return_planet_names
+    planet_names = ""
+    @solar_system.each do |planet_name|
+      planet_names += planet_name.planets_name
+    end
+    return planet_names
+  end
+
+  def planet_details
+    planet_details = ""
+    @solar_system.each do |planet_char|
+      planet_details = planet_char.planet_detail
+    end
+    return planet_details
+  end
+end # end of class
+
+planets_wave_2 = [
+  Planet.new("Angela", 15, 20, "no", 1),
+  Planet.new("Tamira", 10, 2, "yes", 2),
+]
```

Comment:
I like that you put your collection of planets directly into an array here rather than assigning each to its own variable. That makes it clear to the reader that the planets will only be used through the collection, not individually, and saves you having to come up with variable names.